### DISCUSSION TITLE
Acesso a dados Dapper - One to many - Entitadade com mais de uma lista

### DISCUSSION THREAD

**Discussion by rafaelnc2** [balta-io/forum#175] [Tira Dúvidas] [answered]
Balta, 

Caso eu tenha mais de uma lista dentro de uma entidade. Por exemplo, entidade cliente com uma lista de endereços e uma lista de telefones (bem simplista pra facilitar). 

Como ficaria a função que popula os dados? Eu posso criar mais de uma função ou dentro de uma única função eu consigo especificar os dois tipos de lista??

Desde já agradeço a atenção.

**Comment by andrebaltieri** [comment, marked as answer]
Precisa incluir os dois itens... só separar por vírgula no `Query<Cliente, Endereco, Telefone>`. Posta o código aí que ajudamos...

**Reply by rafaelnc2** [reply]
Fala Balta, deu certo aqui. Criei um cenário só pra testar mesmo e funcionou.

Segue.... 

` var sql = @"select  cli.Id, cli.Nome, 
                        [end].Id, [end].Endereco, [end].Cidade, [end].Estado,
                        [tel].Id, [tel].Fixo, [tel].Celular
                from    Cliente as cli
                            inner join
                        Endereco as [end]
                            on [end].cliente_id = cli.id
                            inner join
                        Telefone as tel
                            on tel.cliente_id = cli.id";

    var clientes = new List<Cliente>();

    var items = connection.Query<Cliente, Endereco, Telefone, Cliente>(
        sql,
        (cliente, endereco, telefone) => {
            var cli = clientes.FirstOrDefault(c => c.Id == cliente.Id);

            if(cli is null)
            {
                cli = cliente;

                cli.Enderecos.Add(endereco);
                cli.Telefones.Add(telefone);

                clientes.Add(cli);

                return cliente;
            }

            var end = cli.Enderecos.FirstOrDefault(e => e.Id == endereco.Id);
            if(end is null)
                cli.Enderecos.Add(endereco);

            var tel = cli.Telefones.FirstOrDefault(t => t.Id == telefone.Id);
            if(tel is null)
                cli.Telefones.Add(telefone);

            return cliente;

        }, splitOn: "Id, Id, Id"
    );`

Obrigado pela ajuda.

**Reply by andrebaltieri** [reply]
@rafaelnc2 show demais!!! muito obrigado por compartilhar a resposta 💜
